seacrowd/sea_datasets/xquad/xquad.py

```diff
@@ -0,0 +1,157 @@
+            schema="seacrowd_qa",
+            subset_id="{sub}".format(sub=subset),
+        )
+        for subset in subsets
+    ]
+
+    DEFAULT_CONFIG_NAME = "xquad_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {"context": datasets.Value("string"), "question": datasets.Value("string"), "answers": datasets.Features({"answer_start": [datasets.Value("int64")], "text": [datasets.Value("string")]}), "id": datasets.Value("string")}
+            )
+        elif self.config.schema == "seacrowd_qa":
+            features = schemas.qa_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        name_split = self.config.name.split("_")
+        subset_name = name_split[0]
+
+        if subset_name == "xquad":
+            # For 'xquad' subset, download and combine Vietnamese and Thai subsets
+            vi_filepath = dl_manager.download_and_extract(_URLS + "xquad.vi.json")
+            th_filepath = dl_manager.download_and_extract(_URLS + "xquad.th.json")
+            filepaths = [vi_filepath, th_filepath]
+        else:
+            # For specific language subsets
+            filepath = dl_manager.download_and_extract(_URLS + subset_name + ".json")
+            filepaths = [filepath]
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepaths": filepaths,
+                    "split": "train",
+                },
+            )
+        ]
+
+    def _generate_examples(self, filepaths: List[Path], split: str) -> Tuple[int, Dict]:
+        count = 0
+        document_count = 0
+        for filepath in filepaths:
+            with open(filepath, encoding="utf-8") as f:
+                data = json.load(f)
+
+            for paragraphs in data["data"]:
+                for example in paragraphs["paragraphs"]:
+                    context = example["context"]
+                    for qa in example["qas"]:
+                        question = qa["question"]
+                        answers = qa["answers"]
+                        answers_start = [answer["answer_start"] for answer in answers]
+                        answers_text = [answer["text"] for answer in answers]
+
+                        if self.config.schema == "source":
+                            yield count, {
+                                "context": context,
+                                "question": question,
+                                "answers": {"answer_start": answers_start, "text": answers_text},
+                                "id": count,
+                            }
```

Comment:
I see that you're referring to their HF Dataset Loader. My suggestion is to refactor this for `_generate_examples()` and `_split_generators()` to make it load from their HF repo (using `datasets.load_dataset`) rather than copying it to this dataloader implementation

Reply:
Thank you for the suggestion. Can you elaborate more how do we want it to be refactored?
Do we want the call to `datasets.load_dataset('xquad', 'xquad.vi')`, for example, to happen in `_generate_examples()` or `_split_generators()`?

Reply:
yup, it shd be done under `_generate_examples()`. And you can remove the `download_and_extract` part on the `_split_generators()` if you want to do it. I think by implementing this way, we can achieve better code consistency w/ their HF in the long run, since If they decided to refactor it and we don't load it directly from them, we shd adjust the codes it again (rather than keeping as it is).